### DISCUSSION TITLE
Zapret через WAN1 и WAN2 одновременно на Keenetic 1810

### DISCUSSION THREAD

**Discussion by Drundig** [bol-van/zapret#397] [Q&A]
Добрый день! 
У меня установлен запрет на KN-1810, в роутер заходит два WAN (основной интерфейс eth3) и второй преназначенный LAN в  WAN2 (интерфейс eth2.4) как резервный, на основном интерфейсе висят ПК и тел, а на WAN2 висят все телевизоры.
В конфиге прописал
`IFACE_WAN="eth2.4 eth3"`
Но всё ровно обход запрета работает только на основном WAN интерфейсе.
Искал инфу в интернете, как исправить - но готовых решений не нашел, так как особо не разбираюсь в данном вопросе. Давали советы, что где то в фаерволе нужно что-то прописывать, а где что и как, ни кто ни чего конкретного не смог ответить, то ли в роутере, толи в скрепте zapret.
**Прошу помощи в решении данной задачи**, если возможно по подробнее как правильно, что и где нужно прописать?

**Comment by bol-van** [comment]
Посмотреть правила iptables - есть ли переадресация на -o eth2.4
Включить nfqws в --debug режиме. Посмотреть обрабатываются ли пакеты.

Если провайдеры разные, то и стратегии тоже будут разными. Может не работать для другого

Еще может сама прошивка дергать правила и удалять.

**Reply by Drundig** [reply]
Правила посмотрел командой  iptables -L
получил  [Log.txt](https://github.com/user-attachments/files/17027591/Log.txt)
Я повторюс, что не сильно разбираюсь в данном вопросе, но судя по логу интерфейсы здесь указанны как GigabitEthernet
итерфейсов под названием eth2.4 и eth3 в логе нет совсем, есть
GigabitEthernet0/Vlan4
GigabitEthernet1
и к ним обоим применены одинаковые правила, причем в логе отображен только IP основного WAN 192.168.100.22 а второго IP WAN2 в логе совсем нет.

По поводу разных стратегий, если я меняю местами WAN провода, либо в вебморде роутера делаю основным WAN2 а резервным WAN1 то запрет продолжает работать только на ОСНОВНОМ соединении, а цель моя заставить его работать и на резеврном одновременно.

Прошу помощи в решении данной проблемы.

**Reply by bol-van** [reply]
Так ничего не видно 
iptables -S -t mangle

И нет лога от nfqws --debug

**Reply by Drundig** [reply]
Вот запустил `iptables -S -t mangle` теперь [Log2.txt](https://github.com/user-attachments/files/17033110/Log2.txt) интерфейсы eth2.4 и eth3 видны и они идентичны вроде по правилам. 
Как правильно nfqws --debug запустить - не разобрался.

**Reply by numbereleven-a** [reply]
> Вот запустил `iptables -S -t mangle` теперь [Log2.txt](https://github.com/user-attachments/files/17033110/Log2.txt) интерфейсы eth2.4 и eth3 видны и они идентичны вроде по правилам. Как правильно nfqws --debug запустить - не разобрался.

дописываете перед стратой --debug=@/opt/logfile
пример _NFQWS_OPT_DESYNC="--debug=@/opt/zapret/log.txt --dpi-desync=disorder2 --dpi-desync-ttl=4 --dpi-desync-ttl6=2 --dpi-desync-split-pos=1 --wssize 1:6 --dpi-desync-fooling=md5sig"_

**Reply by bol-van** [reply]
Правила iptables есть. Перед запуском nfqws в консоли надо убить другой инстанс.
От этого и permission denied

**Reply by Drundig** [reply]
Вот debug лог [log3.txt](https://github.com/user-attachments/files/17034647/log3.txt) много писать не стал, думаю нет смысла.
повторяю, что обход работает на оновном соединении, у меня складывается впечатление, что резервное соединение WAN2 идет просто мимо zapret

**Reply by bol-van** [reply]
**rawsend: sendto: Operation not permitted**

Проблема вот в этом
Попробуйте запускать nfqws с дополнительным параметром --bind-fix4
И нужен лог опять

**Reply by Drundig** [reply]
Вот кажется сдела 
[log4.txt](https://github.com/user-attachments/files/17084359/log4.txt)
но теперь в логе **rawsend: bind (ignoring): Address not available**

**Reply by bol-van** [reply]
Это не фатальная ошибка. Такое может случаться на проходящем трафике.
Заработало ?
Если нет, надо проверить уходят ли пакеты  от nfqws на корректный интерфейс. То есть на тот WAN, куда идет соединение

**Reply by bol-van** [reply]
Проблемы с отсылкой и выбором правильного интерфейса обычно происходят, когда применяются множественные таблицы маршрутизации и выбор таблиц по ip rule.
Что показывает ip rule ?

**Reply by Drundig** [reply]
К сожалению не заработало, картинки грузятся с ютуб, а видео не грузится.

~ # ip rule
0:      from all lookup local
10:     from all fwmark 0xffffa00 lookup main
100:    from all fwmark 0xffffaaa lookup 10
101:    from all fwmark 0xffffaaa lookup unspec blackhole
102:    from all fwmark 0xffffaab lookup 11
103:    from all fwmark 0xffffaab lookup unspec blackhole
104:    from all fwmark 0xffffaac lookup 12
105:    from all fwmark 0xffffaac lookup unspec blackhole
106:    from all fwmark 0xffffaad lookup 13
107:    from all fwmark 0xffffaad lookup unspec blackhole
108:    from all fwmark 0xffffaae lookup 14
109:    from all fwmark 0xffffaae lookup unspec blackhole
450:    from 192.168.10.32 lookup 74
452:    from 192.168.11.19 lookup 76
32766:  from all lookup main
32767:  from all lookup default
~ #
Второй WAN это 192.168.10.32 пакеты в логе с этим IP имеются.

**Reply by bol-van** [reply]
к googlevideo идет обращение с 192.168.1.143
Надо tcpdump-ом снять лог с eth2.4, чтобы убедиться, что все идет туда как надо при таких операциях :

```
packet: id=815 len=669 mark=0FFFFAAC ifin=(0) ifout=eth2.4(38)
IP4: 192.168.1.143 => 173.194.181.150 proto=tcp ttl=63 sport=38554 dport=443 flags=AP seq=64883628 ack_seq=3304204092
TCP: 16 03 01 02 64 01 00 02 60 03 03 DC 50 DE 80 E8 55 A9 EB E2 B9 6A 27 6E 67 11 68 24 A9 FF F0 F2 ... : ....d...`...P...U....j'ng.h$.... ...
packet contains full TLS ClientHello
hostname: rr4---sn-n8v7znsk.googlevideo.com
Checking exclude hostlist
Hostlist check for rr4---sn-n8v7znsk.googlevideo.com : negative
Hostlist check for googlevideo.com : negative
Hostlist check for com : negative
Checking include hostlist
Hostlist check for rr4---sn-n8v7znsk.googlevideo.com : negative
Hostlist check for googlevideo.com : positive
dpi desync src=192.168.1.143:38554 dst=173.194.181.150:443
split pos 2
sending 1st tcp segment 0-1 len=2 seqovl=0 : 16 03 : ..
rawsend: bind (ignoring): Address not available
sending 2nd tcp segment 2-616 len=615 : 01 02 64 01 00 02 60 03 03 DC 50 DE 80 E8 55 A9 EB E2 B9 6A 27 6E 67 11 68 24 A9 FF F0 F2 A9 D3 ... : ..d...`...P...U....j'ng.h$...... ...
rawsend: bind (ignoring): Address not available
packet: id=815 drop

```

**Reply by bol-van** [reply]
Судя по марку оно должно идти на таблицу 12.

Что вообще в таблицах ?
ip route show table 10
ip route show table 11
ip route show table 12
ip route show table 13
ip route show table 14
ip route show table 74
ip route show table 76

**Reply by bol-van** [reply]
Тааак.
nfqws на генерируемые пакеты выставляет бит в марк 0x40000000

И это ломает весь ip rule
То есть настройки прошивки , сделанные для multiwan, несовместимы с логикой nfqws
Там надо было использовать не полностью mark, а битовые поля.

Говорю же, keenetic я не поддерживаю. Бороться со стоками и производителем не хочу.
По идее можно поправить ip rule, чтобы заработало, но потом надо следить, чтобы оно на лету не выправило назад.

**Reply by Drundig** [reply]
Вобщем понятно, что мне ни чего не понятно, спасибо за помощь, видимо на кенетике не получится получить с разных WAN обход запрета.

**Reply by Drundig** [reply]
ip route show table 10
default
192.168.10.0/24 dev eth3 scope link
192.168.11.0/24 dev eth2.4 scope link
192.168.1.0/24 dev br0 scope link
~ # ip route show table 11
default via 192.168.10.1 dev eth3
192.168.10.0/24 dev eth3 scope link
192.168.11.0/24 dev eth2.4 scope link
192.168.1.0/24 dev br0 scope link
~ # ip route show table 12
default via 192.168.11.1 dev eth2.4
192.168.10.0/24 dev eth3 scope link
192.168.11.0/24 dev eth2.4 scope link
192.168.1.0/24 dev br0 scope link
~ # ip route show table 13
192.168.10.0/24 dev eth3 scope link
192.168.11.0/24 dev eth2.4 scope link
192.168.1.0/24 dev br0 scope link
~ # ip route show table 14
192.168.10.0/24 dev eth3 scope link
192.168.11.0/24 dev eth2.4 scope link
192.168.1.0/24 dev br0 scope link
~ # ip route show table 74
default via 192.168.10.1 dev eth3  src 192.168.10.32
192.168.10.0/24 dev eth3 scope link  src 192.168.10.32
192.168.10.0/24 dev eth3 scope link
192.168.11.0/24 dev eth2.4 scope link
192.168.1.0/24 dev br0 scope link
~ # ip route show table 76
default via 192.168.11.1 dev eth2.4  src 192.168.11.19
192.168.10.0/24 dev eth3 scope link
192.168.11.0/24 dev eth2.4 scope link  src 192.168.11.19
192.168.11.0/24 dev eth2.4 scope link
192.168.1.0/24 dev br0 scope link

**Reply by bol-van** [reply]
Получится, но средств "для пользователя" для этого нет. А я этим заниматься не стану
